rfc/updates.md

```diff
@@ -0,0 +1,199 @@
+
+The agent will expose the following service:
+```proto
+service Update {
+    rpc ExecuteUpdate(ExecuteUpdateRequest) returns (ExecuteUpdateResponse);
+}
+
+message ExecuteUpdateRequest {
+    string kubeadm_URL = 1;
+    string kubeadm_Hash = 2;
+    string wanted_Kubernetes_Version = 3;
+}
+
+message ExecuteUpdateResponse {
+}
+```
+
+The dependency and usage of the upgrade agent by the node operator is explained in the next section.
+
+
+## Extending the node operator
+
+We need to extend the node operator to also handle Kubernetes updates. The operator already receives information about the Kubernetes version of each node. 
+
+The CLI hands uses the same mechanism to deliver the Kubernetes version to the operator as we [currently use for the image reference](https://github.com/edgelesssys/constellation/blob/main/operators/constellation-node-operator/api/v1alpha1/nodeimage_types.go#L14):
+
+
+```patch
+// NodeImageSpec defines the desired state of NodeImage.
+-type NodeImageSpec struct {
++type NodeSpec struct {
+	// ImageReference is the image to use for all nodes.
+	ImageReference string `json:"image,omitempty"`
++   // KubernetesVersion defines the Kubernetes version for all nodes.
++   KubernetesVersion string `json:"kubernetesVersion,omitempty"`
+}
+```
+
+Additionally, we will change the `NodeImageStatus` to `NodeStatus` (see `nodeimage_types.go`) along with the corresponding controllers. 
+
+The Controller will need to take the following steps to update the Kubernetes version:
+* disable autoscaling 
+* get the kubeadm download URL and hash from the `k8s-components-1.23.12` ConfigMap
+* pass the URL and hash over a socket mounted into its container to the local update agent running on the same node
+    * The agent downloads the new kubeadm binary, checks its hash and executes `kubeadm upgrade plan` and `kubeadm upgrade apply v1.23.12`   
+* After the agent returned successfully, update the Kubernetes version to `1.23.12` and components reference to `k8s-components-1.23.12` in the `k8s-version` ConfigMap 
+* Now, iterate over all nodes, and replace them if their Kubernetes version is outdated
+
+
+## Extending the `constellation upgrade` command
+
+Currently, `constellation upgrade` allows us to upgrade the VM image via the following entry in the constellation-config.yaml:
+
+```yaml
+upgrade:
+  image: /communityGalleries/ConstellationCVM-b3782fa0-0df7-4f2f-963e-fc7fc42663df/images/constellation/versions/2.3.0
+  measurements:
+    11: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=
+    12: AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=
+```
+
+Instead of having a separate `upgrade` section, we will opt for a declarative approach by updating the existing values of the config file. 
```

Comment:
The original idea behind having a separate upgrade section was to discourage users from accidentally trying to change config values which can not be upgrade, e.g. increasing state disk size or VM instance types.

If we want to integrate upgradeable values into the main config, it should be made clear to the users which values are safe to change, and which are not.
Ideally users dont have to interact with their config file at all, and instead are able to use just the CLI commands to perform upgrades

Reply:
> If we want to integrate upgradeable values into the main config, it should be made clear to the users which values are safe to change, and which are not.

Do you have a implementation in mind that would make it clear(er)? We could move all "update-able" values in their own group. 

> Ideally users dont have to interact with their config file at all, and instead are able to use just the CLI commands to perform upgrades

If the user is not supposed to touch the config file, how should Constellation be integrated in an GitOps style deployment pipeline?